content/200-concepts/100-components/02-prisma-client/260-client-extensions/130-result.mdx

```diff
@@ -1,16 +1,148 @@
+    user: {
+      fullName: {
+        needs: { firstName: true, lastName: true }, // the dependencies
+        compute(user) {
+          // the computation logic
+          return `${user.firstName} ${user.lastName}`
+        },
+      },
+    },
+  },
+})
+
+const user = await xprisma.user.findFirst()
+
+// return the user's full name, such as "John Doe"
+console.log(user.fullName)
+```
+
+In above example, the input `user` is automatically typed according to the object defined in `needs`. `firstName` and `lastName` are of type `string`, because they are specified in `needs`. If they are not specified in `needs`, then their type is `string | undefined`.
+
+When you call a method in an extension, use the constant name from your`$extends`statement, not `prisma`. Instead, use the constant name from your`$extends`statement. In the above example,`xprisma.user.findFirst` works, but `prisma.user.findFirst` does not.
+
+### Add a computed field to query results
+
+The following example computes a user's title and full name in a type-safe way.
+
+```ts
+const xprisma = prisma
+  .$extends({
+    result: {
+      user: {
+        fullName: {
+          needs: { firstName: true, lastName: true },
+          compute(user) {
+            return `${user.firstName} ${user.lastName}`
+          },
+        },
+      },
+    },
+  })
+  .$extends({
+    result: {
+      user: {
+        titleFullName: {
+          needs: { title: true, fullName: true },
+          compute(user) {
+            return `${user.title} (${user.fullName})`
+          },
+        },
+      },
+    },
+  })
+```
+
+#### Considerations for fields
+
+- For performance reasons, Prisma Client computes results when your application accesses them at runtime, not on retrieval.
+- You can only create fields that are based on scalar fields.
+- For the `result` component, Prisma Client only computes a field if its dependencies are present. In the following example, `undefined` is returned for `fullName` if the user's `email` is undefined.
+  ```ts
+  const user = await xprisma.user.findFirst({ select: { email: true }))
+  console.log(user.fullName) // undefined
+  ```
+
+### Add a custom method to the result object
+
+You can use the `result` component to add fields to query results. The following example adds a new method, `getAccountBalance` to the result object.
+
+```ts
+const xprisma = prisma.$extends({
+  result: {
+    user: {
+      getAccountBalance: {
+        needs: { account: true },
+        compute((user) => (n: number) =>) {
+          return user.account.balance + n
+        },
+      },
+    },
+  },
+})
+
+const result = await xprisma.user.findFirst()
+console.log(result?.getAccountBalance(-500)) // 7245
+```
+
+When you call a method in an extension, use the constant name from your`$extends`statement, not `prisma`. Instead, use the constant name from your`$extends`statement. In the above example,`xprisma.user.findFirst` works, but `prisma.user.findFirst` does not.
```

Comment:
```suggestion
When you call a method in an extension, use the constant name from your `$extends` statement, not `prisma`. Instead, use the constant name from your `$extends` statement. In the above example,`xprisma.user.findFirst` will return an extra field while `prisma.user.findFirst` won't.
```

Reply:
I'm going to leave this as-is, because where possible we should use present simple tense in docs